base/genimport/importer.go

```diff
@@ -269,14 +274,93 @@ func removeAllFilesInDirExcept(o *Output, dir string, except_list []string) {
 }
 
 func createPluginGoModFile(o *Output, pkgpath string, dir string) string {
+	newLines := [][]byte{
+		[]byte("module gomacro.imports/" + pkgpath + "\n"),
+	}
+
+	// TODO: Figure out a better way to guess the dir
+	if pkgModFileInfo, err := getModuleFileInfo("."); err == nil &&
+		strings.Contains(strings.ToLower(pkgpath), strings.ToLower(pkgModFileInfo.Path)) {
+
+		o.Debugf("importing %s from local %s", pkgpath, pkgModFileInfo.GoMod)
+		newLines = append(newLines, goModReplacementDirectives(o, pkgModFileInfo)...)
+	}
+
+	newGoModText := bytes.Join(newLines, nil)
+
 	gomod := paths.Subdir(dir, "go.mod")
-	err := ioutil.WriteFile(gomod, []byte("module gomacro.imports/"+pkgpath+"\n"), os.FileMode(0644))
+	err := ioutil.WriteFile(gomod, newGoModText, os.FileMode(0o644))
 	if err != nil {
 		o.Errorf("error writing file %q: %v", gomod, err)
 	}
+
 	return gomod
 }
 
+// goModReplacementDirectives will create the replacement directives associated
+// with a module that can be found locally, for the purpose of use in a
+// gomacro.imports mod file.
+func goModReplacementDirectives(o *Output, pkgModFileInfo modInfo) [][]byte {
+	m, err := getModuleFile(pkgModFileInfo)
+	if err != nil {
+		o.Errorf("error getting go.mod", err)
+		return [][]byte{}
+	}
+
+	// Replace the current module with the right directory
+	replacementDirectives := [][]byte{
+		[]byte("replace "+ pkgModFileInfo.Path+" => " + pkgModFileInfo.Dir + "\n"),
+	}
+
+	for _, repplaceDirective := range m.Replace {
+		// TODO: this doesn't work if the replace directives found are themselves local to the repo
+		replacementDirectives = append(replacementDirectives,
```

Comment:
nit: let's be consistent about whether it's a replace directive or a replacement directive (ideally the former since that's what Go calls it)
```suggestion
		replaceDirectives = append(replaceDirectives,
```

Reply:
Good idea, will do!